### DISCUSSION TITLE
How can I ensure isolation between multiple browsers?

### DISCUSSION THREAD

**Discussion by MuuXB** [zauberzeug/nicegui#2257] [Q&A] [answered]
### Question

When using NiceGUI to draw pages (such as charts), the operations between multiple browsers are affecting each other. How can I ensure isolation between multiple browsers?

**Comment by thetableman** [comment, marked as answer]
It sounds like you need to implement pages using the `@ui.page` decorator.

See the FAQs here for links to the documentation: https://github.com/zauberzeug/nicegui/wiki/FAQs#why-are-all-browser-tabswindows-showing-the-same-content
